downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md

```diff
@@ -49,10 +49,64 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 10 (2201.10.0)
 
 ## Runtime updates
 
-### New features
-
 ### Improvements
 
+#### Added support for the `shift`, `unshift`, `pop`, and `remove` functions with tuples
+The runtime now supports the `shift`, `unshift`, `pop`, and `remove` operations on tuples similar to arrays, as long as they do not violate inherent type or mutability constraints.
+
+The following are now allowed.
+```ballerina
+[int, int...] tuple1 = [1, 2];
+int val1 = tuple1.shift();          // 1
+
+[string, string...] tuple2 = ["hello"];
+tuple2.unshift("world", "cat");     // ["hello", "world", "cat"]
+
+[int, string, int...] tuple3 = [1, "hello", 4];
+var val3 = tuple3.pop();            // 4
+
+[int,string, float...] tuple4 = [7, "hello", 67.5, 89.7];
+var val4 = tuple4.remove(2);        // 67.5
+```
+
+The following examples will cause errors.
+```ballerina
+[int, string...] tuple1 = [1, "hello"];
+int val1 = tuple1.shift();          // inherent type violation
+
+[int, int] tuple2 = [1, 2];
+int val2 = tuple2.shift();          // immutable tuple
```

Comment:
I don't think "immutable tuple" is correct here, still inherent type violation.

Reply:
Check the other one too.

By (im)mutabilitity, meant trying to update an immutable value.

Reply:
You are right. Fixed with 978f9c1b83e79a662502bc2dc14d8ac371386ab3. Thank you for pointing that out.